### PR TITLE
Added more features

### PR DESCRIPTION
**Pull request by makermelissa**
This includes more features such as chmod, chown, and the ability for pattern_search to return the result.

### CODE REVIEW

File: adafruit_shell.py [hunk 1]
```diff
@@ -369,6 +373,31 @@ def copy(self, source, destination):
                     destination += os.sep + os.path.basename(source)
                 shutil.copy(source, destination)
 
+    def chmod(self, location, mode):
```
Comment (brentru) [inline adafruit_shell.py:376]: May want to verify `mode` type, https://docs.python.org/3.11/library/os.html#os.chmod

File: adafruit_shell.py [hunk 2]
```diff
@@ -369,6 +373,31 @@ def copy(self, source, destination):
...
+        """
+        Change the permissions of a file or directory
+        """
+        location = self.path(location)
+        if os.path.exists(location):
+            os.chmod(location, mode)
+
+    def chown(self, location, user, group=None, recursive=False):
```
Comment (brentru) [inline adafruit_shell.py:384]: May want to match kwargs with the Python `chown`, https://docs.python.org/3.11/library/os.html#os.chown

**Review by brentru — APPROVED** [review]
@makermelissa  Reviewed, left two points of optional feedback (not required to merge)

File: adafruit_shell.py (hunk 2 above)
Comment (makermelissa) [reply adafruit_shell.py:384]: Good idea. Thanks.

File: adafruit_shell.py (hunk 1 above)
Comment (makermelissa) [reply adafruit_shell.py:376]: Yeah, I wanted either an octal value in the range of 0o0 to 0o777 or the flags (which should result in an octal value in the same range), so at least a value range is a good check to add.

File: adafruit_shell.py (hunk 2 above)
Comment (makermelissa) [reply adafruit_shell.py:384]: It looks like there's nothing additional for the shutil.chown function: https://docs.python.org/3/library/shutil.html#shutil.chown.

**Pull request merged**